telegram/ext/_picklepersistence.py

```diff
@@ -17,23 +17,119 @@
 # You should have received a copy of the GNU Lesser Public License
 # along with this program.  If not, see [http://www.gnu.org/licenses/].
 """This module contains the PicklePersistence class."""
+import copyreg
 import pickle
+from copy import deepcopy
 from pathlib import Path
+from sys import version_info as py_ver
 from typing import (
     Any,
     Dict,
     Optional,
     Tuple,
     overload,
     cast,
+    Type,
+    Set,
+    Callable,
+    TypeVar,
 )
 
+from telegram import Bot, TelegramObject
 from telegram._utils.types import FilePathInput
+from telegram._utils.warnings import warn
 from telegram.ext import BasePersistence, PersistenceInput
 from telegram.ext._contexttypes import ContextTypes
 from telegram.ext._utils.types import UD, CD, BD, ConversationDict, CDCData
 
 
+_REPLACED_KNOWN_BOT = "a known bot replaced by PTB's PicklePersistence"
+_REPLACED_UNKNOWN_BOT = "an unknown bot replaced by PTB's PicklePersistence"
+
+TO = TypeVar('TO', bound=TelegramObject)
+
+
+def _all_subclasses(cls: Type[TO]) -> Set[Type[TO]]:
+    """Gets all subclasses of the specified object, recursively. from
+    https://stackoverflow.com/a/3862957/9706202
+    """
+    subclasses = cls.__subclasses__()
+    return set(subclasses).union([s for c in subclasses for s in _all_subclasses(c)])
+
+
+def _reconstruct_to(cls: Type[TO], kwargs: dict) -> TO:
+    """
+    This method is used for unpickling. The data, which is in the form a dictionary, is
+    converted back into a class. Functions the same as :meth:`TelegramObject.__setstate__`.
```

Comment:
```suggestion
    converted back into a class. Works mostly the same as :meth:`TelegramObject.__setstate__`.
```

Why not use `TO.__setstate__`, then?

Reply:
there's a subtle difference, `__setstate__` seems to construct a new instance internally and should not return anything (modify in place is only allowed). Here, we have to construct a new instance manually. Plus, `TO.__setstate__` assigns to `self`, here we need it to assign to our new instance (`obj`)

Reply:
what I have in mind is something like 

```python
obj = cls.__new__(cls)
obj.__setstate__(kwargs)
```

IISC that should work just fine